### ISSUE TITLE
Multi gpus training bug

### ISSUE THREAD

**Issue by zezeze97** [jayleicn/ClipBERT#37]
Hi, thanks for sharing the code.  I encountered bug when I tried to make a finetuning on the VQA2.0 dataset. The logs are displayed as follows. The resulting accuracy is only 60, far short of what you did in your paper. I'm training with 2*2080ti graphics cards.

[1,0]<stderr>:[2021-11-20 04:52:04.885367: W horovod/common/stall_inspector.cc:105] One or more tensors were submitted to be reduced, gathered or broadcasted by subset of ranks and are waiting for remainder of ranks for more than 60 seconds. This may indicate that different ranks are trying to submit different tensors or that only subset of ranks is submitting tensors, which will cause deadlock. 
[1,0]<stderr>:Stalled ranks:
[1,0]<stderr>:0: [allreduce.cnn.feature.backbone.res3.0.conv1.weight, allreduce.cnn.feature.backbone.res3.0.conv2.weight, allreduce.cnn.feature.backbone.res3.0.conv3.weight, allreduce.cnn.feature.backbone.res3.0.shortcut.weight, allreduce.cnn.feature.backbone.res3.1.conv1.weight, allreduce.cnn.feature.backbone.res3.1.conv2.weight ...]
[1,0]<stderr>:[2021-11-20 04:53:04.888208: W horovod/common/stall_inspector.cc:105] One or more tensors were submitted to be reduced, gathered or broadcasted by subset of ranks and are waiting for remainder of ranks for more than 60 seconds. This may indicate that different ranks are trying to submit different tensors or that only subset of ranks is submitting tensors, which will cause deadlock. 
[1,0]<stderr>:Stalled ranks:
[1,0]<stderr>:0: [allreduce.cnn.feature.backbone.res3.0.conv1.weight, allreduce.cnn.feature.backbone.res3.0.conv2.weight, allreduce.cnn.feature.backbone.res3.0.conv3.weight, allreduce.cnn.feature.backbone.res3.0.shortcut.weight, allreduce.cnn.feature.backbone.res3.1.conv1.weight, allreduce.cnn.feature.backbone.res3.1.conv2.weight ...]

**Comment by jayleicn** [comment]
I am not sure how this error happens. For your lower accuracy, it might because that your setting differs from ours, e.g., you used fewer GPUs to train the model.

**Comment by linjieli222** [comment]
`stalled ranks` means that the different processes are not synced, hence one or more of them get stalled waiting for the other ranks. We also observed `stalled ranks` from time to time, this is  a warning message from horovod implementation, not from our code. Therefore, we do not have an immediate solution. Note that `stalled ranks` may cause the training stay idle, but if the experiment is still running when you see this warning message, the final performance should not be affected. 

As mentioned by @jayleicn , the accuracy difference might be related to the number of GPUs, as our hyperparameters search is based on 4 V100 GPUs. 
One possible way to reach similar accuracy is to increase the gradient accumulation steps. For example, double the gradient accumulation steps if you are using only two GPUs. And I suspect you also need to decrease the batch size to accommodate for 2080ti GPUs (11GB vs. 32GB used in our experiments). These are the factors that may impact the performance.
